zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingScheduleServiceImpl.java

```diff
@@ -180,4 +209,25 @@ Runnable toRunnable(final Task task) {
           });
     };
   }
+
+  /** Note: this class has a natural ordering that is inconsistent with equals. */
+  private final class ScheduledTaskImpl implements ScheduledTask, Comparable<ScheduledTaskImpl> {
```

Comment:
:question: Would making it a record reduce its memory footprint in the priority queue? I know the actual memory difference is generally small, but may be more at scale - and I believe (I'd have to look up which Java quark went into it) that the GC handles records more efficiently, so it might be a better choice overall for storing potentially many of them in an unbounded collection.

Reply:
I can't find a source for this. The only benefit I'm aware of is that the JVM trusts that record fields are immutable (unlike regular object fields) and can apply more optimizations that way.

I've deliberately chosen not to use a record because it's not just holding state but allows cancellation too.


I'm happy to change it to a record though if you can find a source for memory/gc benefits :)